include/finufft/utils.h

```diff
@@ -14,51 +14,44 @@ namespace utils {
 
 // ahb's low-level array helpers
 template<typename T>
-FINUFFT_EXPORT T FINUFFT_CDECL relerrtwonorm(BIGINT n, std::complex<T> *a,
-                                             std::complex<T> *b)
+FINUFFT_EXPORT T FINUFFT_CDECL relerrtwonorm(BIGINT n, const std::complex<T> *a,
+                                             const std::complex<T> *b)
 // ||a-b||_2 / ||a||_2
 {
   T err = 0.0, nrm = 0.0;
   for (BIGINT m = 0; m < n; ++m) {
-    nrm += real(conj(a[m]) * a[m]);
-    std::complex<T> diff = a[m] - b[m];
-    err += real(conj(diff) * diff);
+    nrm += std::norm(a[m]);
```

Comment:
`std::norm(a)` is defined as `abs(a)**2` in C++; it feels unintuitive to me, but I think this is standard in the mathematical community.

Reply:
I got curious now and did a bit of digging :-)
`std::norm` claims to be the field norm, which justifies the square.
From a technical standpoint I totally understand this choice, since it avoids the (comparatively expensive) square root and produces a quantity that is needed very often (also `std::abs` already exists if you want the "normal" norm).
I'm using the function often, since it it has just 3 arithmetic operations compared to the (theoretical) 7 operations (+ one copy) of `(a*conj(a)).real()`, but the optimizer will most likely produce the same code for both.